include/deal.II/base/aligned_vector.h

```diff
@@ -465,9 +465,215 @@ class AlignedVector
 
 private:
   /**
-   * Pointer to actual data array.
+   * A class that is used as the "deleter" for a `std::unique_ptr` object that
+   * AlignedVector uses to store the memory used for the elements.
+   *
+   * There are two ways the AlignedVector class can handle memory:
+   * - Allocation via `new[]` in reserve() where we call `posix_memalign()`
+   *   to obtain a chunk of memory and then do placement-`new` to initialize
+   *   memory. If this is what we have done, then we need to call the
+   *   destructors of the currently active elements by hand, and then
+   *   call `std::free()` to return memory. In order to call the destructors
+   *   of currently used elements, the deleter object needs to have access
+   *   to the owning `AlignedVector` object to know which of the allocated
+   *   elements are currently actually used.
+   * - We have called `replicate_across_communicator()`, in which case the
+   *   elements have been moved into a memory "window" managed by MPI.
+   *   In that case, one process (the root process of an MPI communicator
+   *   that ties together all processes on one node) needs to call the
+   *   destructor of all elements in this shared memory space, and then
```

Comment:
That's an interesting observation one could pursue in a follow-up. How does one test this with a `static_assert`? *Every* class has a destructor, it might just be the default-generated one, but even in that latter case it just calls the (possibly nontrivial) destructors of the members.

Reply:
I checked the list of `std::is_...` trait classes at https://en.cppreference.com/w/cpp/header/type_traits but I can't quite find what I'm looking for. Both `std::is_trivially_copyable` and `std::is_pod` seemed reasonable to me, but they are both `true` for structures that contain pointers.

Reply:
I would go with something like is_trivial. You could have objects with pointers still of course.